### PR TITLE
verify linked transaction for get request

### PR DESCRIPTION
**Pull request by swalke-ch**
Added functionality for verify linked transaction

https://companieshouse.atlassian.net/browse/IDVA5-787

### Type of change

* [ ] Bug fix
* [x ] New feature
* [ ] Breaking change
* [ ] Infrastructure change

### Pull request checklist

* [ x] I have added unit tests for new code that I have added
* [ x] I have added/updated functional tests where appropriate
* [ x] The code follows our [coding standards](https://github.com/companieshouse/styleguides/blob/master/java.md)

__An exhaustive list of peer review checks can be read [here](https://github.com/companieshouse/styleguides/blob/master/java_review.md#developer-actions-prior-to-code-commitreview-started)__

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by ttingle-ch — APPROVED** [review]
LGTM

File: src/main/java/uk/gov/companieshouse/acsp/controllers/AcspController.java [hunk 1]
```diff
@@ -39,7 +43,14 @@ public ResponseEntity<Object> getAcspData(@PathVariable(TRANSACTION_ID_KEY) Stri
                                               @PathVariable("id") String id,
                                               @RequestHeader(value = ERIC_ACCESS_TOKEN) String requestId) {
         LOGGER.info("received request to get acsp data");
-        var acspData = acspService.getAcsp(id);
+        Transaction transaction;
```
Comment (sampankumar) [inline src/main/java/uk/gov/companieshouse/acsp/controllers/AcspController.java:46]: You can declare this inside the try block only using var

File: src/main/java/uk/gov/companieshouse/acsp/service/FilingsService.java [hunk 2]
```diff
@@ -66,18 +67,19 @@ public FilingApi generateAcspApplicationFiling(
...
 
   private void setFilingApiData(FilingApi filing, String acspApplicationId, String transactionId,
-                                String passThroughTokenHeader) throws ServiceException {
-    var acspDataDto = acspService.getAcsp(acspApplicationId).orElse(null);
+                                String passThroughTokenHeader) throws ServiceException, SubmissionNotLinkedToTransactionException {
+    var transaction = transactionService.getTransaction(passThroughTokenHeader, transactionId);
+    var acspDataDto = acspService.getAcsp(acspApplicationId, transaction).orElse(null);
     if(acspDataDto != null) {
       var data = new HashMap<String, Object>();
-      var transaction = transactionService.getTransaction(passThroughTokenHeader, transactionId);
```
Comment (sampankumar) [inline src/main/java/uk/gov/companieshouse/acsp/service/FilingsService.java:80]: I think the purpose of this being inside if was do this only if DTO is not null

Comment (swalke-ch) [reply src/main/java/uk/gov/companieshouse/acsp/service/FilingsService.java:80]: We need Transaction to fetch the DTO

File: src/main/java/uk/gov/companieshouse/acsp/controllers/AcspController.java (hunk 1 above)
Comment (swalke-ch) [reply src/main/java/uk/gov/companieshouse/acsp/controllers/AcspController.java]: moved it to inside try catch block. Kept the Type same as we know the return type will be transaction.

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/main/java/uk/gov/companieshouse/acsp/controllers/AcspController.java (hunk 1 above)
Comment (sampankumar) [reply src/main/java/uk/gov/companieshouse/acsp/controllers/AcspController.java]: I think this adds a code smell.

File: src/main/java/uk/gov/companieshouse/acsp/service/FilingsService.java (hunk 2 above)
Comment (sampankumar) [reply src/main/java/uk/gov/companieshouse/acsp/service/FilingsService.java:80]: Sorry missed that. Cool

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by sampankumar — APPROVED** [review]

**Review by ttingle-ch — APPROVED** [review]

**Pull request merged**